full-service/src/service/password_manager.rs

```diff
@@ -0,0 +1,440 @@
+// Copyright (c) 2020 MobileCoin Inc.
```

Comment:
@eranrund - I also refactored the password management functionality that was originally in the wallet_service.impl into its own service trait. LMK what you think about this pattern, because I'd like to use it going forward with gift codes, and eventually refactor the other logical services out of wallet_service.impl.

Reply:
I think this pattern is good. I will review the file more closely once the _hash stuff goes away.